src/index.js

```diff
@@ -37,22 +37,20 @@ function installFiles(sourceDir, done) {
   switch (npmv.majorVersion()) {
     case '1':
       console.log("[install-files]: WARNING: NPMv1 is not officially supported; unexpected results could occur. Consider upgrading to v2 or later");
+      /* falls through */
     case '2':
       source = sourceDir;
       target = fileInstallingPackagePath && hostPackageDir(fileInstallingPackagePath);
       break;
     case null:
-      console.log("[install-files]: WARNING: Could not determine NPM version"); //Fall back to default
+      console.log("[install-files]: WARNING: Could not determine NPM version");
+      /* falls through */
     default:
       source = path.join(fileInstallingPackagePath, 'node_modules', process.env.npm_package_name, sourceDir);
-      target = fileInstallingPackagePath
+      target = fileInstallingPackagePath;
   }
 
-  if (fileInstallingPackagePath.match(".+" + process.env.npm_package_name + "$")) {
```

Comment:
@wearhere Actually, sorry about the previous comment.  I meant I would have wanted to check `target.match(...)`. My intentions here were to catch a case where the project that hosts the shared files runs `npm install`.  In this case, the process was running against itself, which was not ideal (but in thinking about it now, in some scenarios, this could be necessary).  I would say a better cross-version solution here would be to check to see if the source and the target package names are the same, and reject if that is the case. Thoughts?